common/crypto/openssl/init.c

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/crypto/init.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+
+#ifdef OE_BUILD_ENCLAVE
+#include <openenclave/internal/thread.h>
+#include <openenclave/internal/trace.h>
+#include <openssl/engine.h>
+static oe_once_t _openssl_initialize_once;
+static ENGINE* eng;
+#else
+#include <pthread.h>
+static pthread_once_t _once = PTHREAD_ONCE_INIT;
+#endif
+
+#ifdef OE_BUILD_ENCLAVE
+static void _finalize(void)
```

Comment:
Move OE_BUILD_ENCLAVE code to the enclave subdir instead of under the common subdir.